collaboration/sig-core/proposals/upgrade-to-knative-mesh.md

```diff
@@ -0,0 +1,102 @@
+# Proposal: migration to Knative Event Mesh
+
+## Contents
+
+1. [Expectation](#expectation)
+2. [Overview](#overview)
+3. [First Kyma upgrade : 1.11](#first-kyma-upgrade--111)
+     - [I. Missing HttpSource objects](#i-missing-httpsource-objects)
+     - [II. Legacy endpoints compatibility](#ii-legacy-endpoints-compatibility)
+     - [III. User namespace migration](#iii-user-namespace-migration)
+4. [Second Kyma upgrade : 1.12](#second-kyma-upgrade--112)
+     - [I. Purge event-bus component](#i-purge-event-bus-component)
+
+## Expectation
+
+At the end of the Kyma upgrade, the user is left with an event mesh that has feature parity with the previous Kyma
+version, without the need for any manual configuration and without message loss.
+
+## Overview
+
+The upgrade happens in two steps over two Kyma releases to avoid creating a major disruption. In
+the first step, we ensure Kyma migrated totally to the new Knative Eventing mesh and that `Event Bus` is no longer
+functioning. In the second step, we remove leftovers from all other charts and
+delete the `event-bus` release in a manual step documented in Kyma upgrade guide.
+
+<!--
+described below are orchestrated by the Kyma operator. During a Kyma upgrade, the operator proceeds
+component by component, iteratively. The most rational place to hook our migration logic is at the level of the
+`cluster-essentials` component, which is the [very first chart](./installation/resources/installer-cr.yaml.tpl#L13-L15)
+defined in the `Installation` object, so we can clear the path from there for the actual components' upgrades.
+-->
+
+Each of the steps described below is implemented as a Kubernetes `Job` triggered by a `pre-upgrade` Helm hook. A
+predictable ordering is enforced using the `helm.sh/hook-weight` annotation.
+
+A failed step restores the state of the cluster as it was when the step started.
+
+## First Kyma upgrade : 1.11
+
+The goal of this step is that the existing `event-bus` objects have their Knative Eventing mesh equivalence ones and
+that`Event Bus` is no longer running.
+
+### I. Missing HttpSource objects
+
+#### Changes summary
+
+| Name | Description | Artifact |
+|------|-------------|----------|
+|HttpSource sync binary| A Go binary which creates missing HttpSources | Docker image|
+|`application-connector` `pre-upgrade` hook | A `pre-upgrade` Job which will be executed when Kyma installer upgrades the `application-connector` chart |`application-connector` new version |   
+
+#### Execution logic 
+
+- Create an `HttpSource` for each existing `Application` which has no existing one already.
```

Comment:
wouldn't the application operator reconcile all the applications post upgrade and create HTTPSource at the time of reconciliation

Reply:
Yes